File: backend/booking-service/src/main/java/com/ticketbooking/booking/entity/Seat.java
package com.ticketbooking.booking.entity;

import com.ticketbooking.common.audit.AuditableEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "seats")
public class Seat extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private UUID eventId;

    @Column(nullable = false)
    private String section;

    @Column(nullable = false)
    private String row;

    @Column(nullable = false)
    private String number;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SeatState state = SeatState.AVAILABLE;
}

File: frontend/src/types/seat.ts
export type SeatState = 'AVAILABLE' | 'RESERVED' | 'SOLD';

export interface Seat {
  id: string;
  eventId: string;
  section: string;
  row: string;
  number: string;
  state: SeatState;
}

/** Pushed over WebSocket on every seat state change. */
export interface SeatStatusUpdate {
  seatId: string;
  state: SeatState;
}
